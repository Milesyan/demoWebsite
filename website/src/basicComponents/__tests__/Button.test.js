/* eslint-disable no-undef */ /*eslint-disable react/react-in-jsx-scope*/

import { Button } from '../index';

const mockOnClick = jest.fn();

const setup = overrideProps => {
  const initProps = {
    isDisabled: false,
    onClick: mockOnClick
  };
  const props = { ...initProps, ...overrideProps};
  const wrapper = shallow(
    <Button {...props}/>
  );
  return { wrapper, props };
}

describe('Button snapshots', () => {

  it('should match snapshot when button is not disabled', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should match snapshot when button is disabled', () => {
    const { wrapper } = setup({isDisabled: true});
    expect(wrapper).toMatchSnapshot();
  });

});

describe('Button functions', () => {

  it('should call onClick function when the button is not disabled.', () => {
    const { wrapper } = setup();
  });
  
});