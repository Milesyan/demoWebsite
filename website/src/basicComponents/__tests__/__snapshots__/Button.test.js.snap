// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot should match snapshot when button is disabled 1`] = `
<button
  className="primaryBtn disabledBtn"
  onClick={null}
>
  Default Button Text
</button>
`;

exports[`Button Snapshot should match snapshot when button is not disabled 1`] = `
<button
  className="primaryBtn"
  onClick={[MockFunction]}
>
  Default Button Text
</button>
`;

exports[`Button snapshots should match snapshot when button is disabled 1`] = `
<button
  className="primaryBtn disabledBtn"
  onClick={null}
>
  Default Button Text
</button>
`;

exports[`Button snapshots should match snapshot when button is not disabled 1`] = `
<button
  className="primaryBtn"
  onClick={[MockFunction]}
>
  Default Button Text
</button>
`;
